FirebaseAuth/Tests/SampleSwift/AuthenticationExample/ViewControllers/OtherAuthMethodControllers/PhoneAuthViewController.swift

```diff
@@ -74,4 +75,17 @@ class PhoneAuthViewController: OtherAuthViewController {
 
     present(phoneAuthController, animated: true, completion: nil)
   }
+
+  private func getVerificationCode() async throws -> String {
+    return try await withCheckedThrowingContinuation { continuation in
+      self.presentPhoneAuthController { code in
+        if code != "" {
+          continuation.resume(returning: code)
+        } else {
+          // Cancelled
+          continuation.resume(throwing: NSError())
```

Comment:
throw a descriptive error

Reply:
Can you help understand how to throw a descriptive error. The NSError() initializer captures - domain, code and description. I'm not sure about either. 
Ref: https://github.com/firebase/firebase-ios-sdk/blob/2f6ac4c2c61cd57c7ea727009e187b7e1163d613/FirebaseAuth/Tests/SampleSwift/AuthenticationExample/ViewControllers/UserViewController.swift#L256

Reply:
We should use Swift errors instead of NSErrors. And ideally presentPhoneAuthController should be converted to a throwing function.

Reply:
Looks like we have a similar problem here https://github.com/firebase/firebase-ios-sdk/blob/2f6ac4c2c61cd57c7ea727009e187b7e1163d613/FirebaseAuth/Tests/SampleSwift/AuthenticationExample/ViewControllers/UserViewController.swift#L256
Perhaps we can address the implementation issues of viewcontrollers in a different PR for cleanup?